tests/ArgentPonyWarcraftClient.Integration.Tests/GameDataApi/CovenantApiTests.cs

```diff
@@ -0,0 +1,143 @@
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Conduits_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<ConduitsIndex> result = await client.GetConduitsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/conduit/index?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Covenant_Media_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<CovenantMedia> result = await client.GetCovenantMediaAsync(3, "static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/media/covenant/3?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Covenants_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<CovenantsIndex> result = await client.GetCovenantsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/index?namespace=static-us&locale=en_US"
+            );
+        }
+
+        [ResilientFact]
+        public async Task When_Getting_Soulbinds_Index_Then_Successful_Result_With_Expected_Content_Is_Returned()
+        {
+            ICovenantApi client = ClientFactory.BuildClient();
+
+            RequestResult<SoulbindsIndex> result = await client.GetSoulbindsIndexAsync("static-us");
+
+            Assert.NotNull(result);
+            Assert.True(result.Success);
+            Assert.Null(result.Error);
+            Assert.NotNull(result.Value);
+
+            await AssertResultMatchesBlizzardResponseAsync(
+                result,
+                "https://us.api.blizzard.com/data/wow/covenant/soulbind/index?namespace=static-us&locale=en_US"
+            );
+        }
+
+        private async Task AssertResultMatchesBlizzardResponseAsync<T>(RequestResult<T> actualValue, string blizzardRequestUri)
```

Comment:
This is nice, but we'll probably want to move this out of here before all is said and done.

Reply:
Yup! I typically start with keeping things within a class where they're used and gradually refactor to make it more broadly usable (try not to over engineer in the first go, y'know). Plan is to refactor this into a more accessible method so that we can make use of it in other tests as well in a separate PR 👍